src/prefect/client/orion.py

```diff
@@ -981,6 +981,30 @@ async def update_block_document(
             else:
                 raise
 
+    async def replace_block_document(
```

Comment:
Why `replace` instead of `update`?

Reply:
Based off https://github.com/PrefectHQ/prefect/issues/7264

When users try clearing the value from an optional field, it keeps the old value.

Reply:
Ah gee we have an existing `update` that uses PATCH. I see. I wonder if we should just make this a keyword argument for `update_block_document`? I'm hesitant to deviate from our existing naming patterns.

Reply:
If we make it a kwarg, what about the API?  https://github.com/PrefectHQ/prefect/pull/7470/files#diff-f5cd2ecfc6dc2faf3ae2463c457bfb363fd32636208d824c28bd6165dc25b0cbR138-R139

Reply:
We could call the same route with POST instead of PATCH. Just wondering, I don't see any differences from the existing method in the client / API implementations — is that work in progress still? Perhaps once I know what will need to change it'll be clearer.

Reply:
Oh I meant the naming of the function since you mentioned you're hesitant to deviate from our existing naming patterns.

Yeah there's no changes at the moment because I wasn't sure if changing `PATCH` to `POST` would automatically do everything for me, but based on your comment, I guess not. I probably have to add another `replace_block_document` function similar to [`update_block_document`](https://github.com/PrefectHQ/prefect/blob/e7e6808e056d1dd8ba9773073ca4c2b6bbc2bdf5/src/prefect/orion/models/block_documents.py#L421)

Reply:
Maybe I should use the PUT method instead based off https://stackoverflow.com/questions/31089221/what-is-the-difference-between-put-post-and-patch#:~:text=Here%20is%20a%20simple%20description,always%20for%20updating%20a%20resource

Reply:
Ah yeah POST/PATCH is just the type of request, we need to change our behavior accordingly.

I think in the models `update_block_document` can take a `merge_existing_data` bool. PATCH would set it to true and `POST` would set it to false. It's probably safe to reuse the `update_block_document` client method and `BlockDocumentUpdate` action.

Reply:
I think we may be using POST for cases like this elsewhere although a PUT may be semantically correct since we're only updating the `data` field. It's kind of a murky area. cc @zangell44 / @chrisguidry.